subsys/mgmt/mcumgr/grp/os_mgmt/src/os_mgmt.c

```diff
@@ -363,6 +363,53 @@ os_mgmt_mcumgr_params(struct smp_streamer *ctxt)
 }
 #endif
 
+#if defined(CONFIG_MCUMGR_GRP_OS_BOOTLOADER_INFO)
+
+#if defined(CONFIG_MCUBOOT_BOOTLOADER_MODE_SINGLE_APP)
+#define BOOTLOADER_MODE OS_MGMT_BOOTLOADER_INFO_MCUBOOT_MODE_SINGLE_APP
+#elif defined(MCUBOOT_BOOTLOADER_MODE_SWAP_WITHOUT_SCRATCH)
+#define BOOTLOADER_MODE OS_MGMT_BOOTLOADER_INFO_MCUBOOT_MODE_SWAP_WITHOUT_SCRATCH
+#elif defined(MCUBOOT_BOOTLOADER_MODE_SWAP_SCRATCH)
+#define BOOTLOADER_MODE OS_MGMT_BOOTLOADER_INFO_MCUBOOT_MODE_SWAP_SCRATCH
+#elif defined(MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP)
+#define BOOTLOADER_MODE OS_MGMT_BOOTLOADER_INFO_MCUBOOT_MODE_DIRECT_XIP
+#else
+#define BOOTLOADER_MODE OS_MGMT_BOOTLOADER_INFO_MCUBOOT_MODE_UNKNOWN
+#endif
+
+static int
+os_mgmt_bootloader_info(struct smp_streamer *ctxt)
+{
+	zcbor_state_t *zse = ctxt->writer->zs;
+	zcbor_state_t *zsd = ctxt->reader->zs;
+	size_t decoded;
+	int bootloader_mode = OS_MGMT_BOOTLOADER_INFO_MCUBOOT_MODE_UNKNOWN;
+	bool ok;
+
+	struct zcbor_map_decode_key_val bootloader_info[] = {
+		ZCBOR_MAP_DECODE_KEY_DECODER("mode", zcbor_int32_decode, &bootloader_mode),
```

Comment:
Having a think about this, not sure this will work well like this if the parameter name is what is set, if there are e.g. 20 different possible options, that means you need to try decoding 20 values, would be better if there was just one e.g. "option" or "query" which contains a string, that way you only ever need to decode one possible value and then do a string comparison.

Reply:
OK, will change.